### PR TITLE
Add in support for Django 5.0 by adding in compat functionality for datetime_safe

### PR DESCRIPTION
**Pull request by chander**
This change adds in support for Django 5.0 in Tastypie by doing the following:

1.  Add in a compatability function.
2. Update the tests to properly indicate the versions of python and django suitable for testing with this release.
3. Ensure that tox tests pass properly.

> Running Sphinx v7.2.6
> making output directory... done
> building [mo]: targets for 0 po files that are out of date
> writing output... 
> building [doctest]: targets for 51 source files that are out of date
> updating environment: [new config] 51 added, 0 changed, 0 removed
> reading sources... [100%] who_uses
> looking for now-outdated files... none found
> pickling environment... done
> checking consistency... done
> running tests...
> 
> Document: cookbook
> ------------------
> 1 items passed all tests:
>   16 tests in default
> 16 tests in 1 items.
> 16 passed and 0 failed.
> Test passed.
> 
> Doctest summary
> ===============
>    16 tests
>     0 failures in tests
>     0 failures in setup code
>     0 failures in cleanup code
> build succeeded.
> 
> Testing of doctests in the sources finished, look at the results in ../tmp/html/output.txt.
>

### CODE REVIEW

**Comment by chander** [comment]
@georgedorn can you please review and merge to add Django 5.0 support to tastypie?  Thanks

File: tests/settings.py [hunk 1]
```diff
@@ -54,14 +55,18 @@
...
     'django.contrib.auth.hashers.CryptPasswordHasher',
     'django.contrib.auth.hashers.PBKDF2PasswordHasher',
     'django.contrib.auth.hashers.PBKDF2SHA1PasswordHasher',
     'django.contrib.auth.hashers.BCryptPasswordHasher',
     'django.contrib.auth.hashers.SHA1PasswordHasher',
     'django.contrib.auth.hashers.MD5PasswordHasher',
-)
+]
+# Django 5.0 removed this hasher
+if DJANGO_VERSION >= DJANGO_42:
```
Comment (georgedorn) [inline tests/settings.py:67]: Should this be `DJANGO_VERSION > DJANGO_42`, given that it wasn't removed until 5.0?

Comment (chander) [reply tests/settings.py]: Should be good now.

**Comment by chander** [comment]
@georgedorn should be good to go now -unless you have other changes?

**Review by georgedorn — APPROVED** [review]

**Pull request merged**